webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/precomputed/compressedsegmentation/CompressedSegmentation.scala

```diff
@@ -0,0 +1,123 @@
+package com.scalableminds.webknossos.datastore.datareaders.precomputed.compressedsegmentation
+
+import java.nio.{ByteBuffer, ByteOrder}
+import scala.reflect.ClassTag
+
+// Based on https://github.com/janelia-flyem/compressedseg, see LICENSE.txt
+// Original author: Stephen Plaza (plazas@janelia.hhmi.org)
+
+trait CompressedSegmentation[T <: AnyVal] {
+
+  private val kBlockHeaderSize = 2
+  private val defaultBlockSize = Array(8, 8, 8)
+
+  val typeWidthBytes: Int
+
+  def readValue(input: Array[Int], position: Int): T
+
+  def initializeArray(length: Int): Array[T]
+
+  private def decompressChannel(input: Array[Int], volumeSize: Array[Int], blockSize: Array[Int]): Array[T] = {
+    assert(blockSize.length == 3)
+    val numElements = volumeSize(0) * volumeSize(1) * volumeSize(2)
```

Comment:
is the order of volumeSize elements fixed? can we rely on 0,1,2 being x,y,z, and 3 being c? Could there be a time axis as well?

Reply:
The specification only describes 3d volume data, so I think it can be left like this.